Mobile Discover

cuahsimarko opened the pull request:
Deepen mobile and cross-browser support for Discover, while paving the way for areas of HydroShare to reach this elevated standard. Other areas of HydroShare also have pre-existing problems meeting the former, informal standard and that will need to be addressed as a part of a wider effort.

On https://dev-hs-6.cuahsi.org/ using the same login for your production HydroShare account, verify the following:

1. Fixes https://github.com/hydroshare/hydroshare/issues/4033
2. Does the prep work for 3875, 4108 and 4080, 4089 modularized and changed CSS to be rolled out to pages other than Discover later
3. Fixes https://github.com/hydroshare/hydroshare/issues/3698 (or can't reproduce)
4. Fixes https://github.com/hydroshare/hydroshare/issues/3926

Extensive cleanup in the source code performed, a keen eye on functionality to ensure there are no regressions is warranted.

hydrocheck commented:
| METRIC | VALUE |
|:-----:|:-----:|
| NEW BUGS | 15 |
| NEW DUPLICATED LINES DENSITY | 8.65546218487395 |
| NEW VULNERABILITIES | 0 |
| NEW CODE SMELLS | 0 |
| NEW SECURITY HOTSPOTS | 0 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4126](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4126)

hydrocheck commented:
On all window widths in Safari 13 (MacOS Catalina), the table is below the filters.
![Screen Shot 2021-02-11 at 10 43 50 PM](https://user-images.githubusercontent.com/43424907/107728436-12ccdf80-6cbc-11eb-8848-566c87731d5c.png)

cuahsimarko commented:
@hydrocheck I have addressed the legend issue that was an oversight on work I had to redo multiple times moving files around. I also fixed the date column issue, where the last digits were sometimes line wrapping. I am unable to reproduce the Safari issue, can you clear caches and try again? Could also have been a template/deployment quirk from the last deployment.

cuahsimarko commented on hs_discover/public/index.html:
```diff
@@ -10,36 +10,53 @@
         /* a height must be specified to make visible */
         height: 50%;
     }
```
rearrange, placing `#id` at top and `.class` towards bottom

cuahsimarko commented on hs_discover/public/index.html:
```diff
@@ -10,36 +10,53 @@
+    }
+    .vdpCell.today {
+        color: Red;
+    }
```
Most calendars online I skimmed across seem to use red

cuahsimarko commented on hs_discover/public/index.html:
```diff
@@ -10,36 +10,53 @@
-         width: 233px;
+    .vdpCell.selectable:hover .vdpCellContent, .vdpCell.selected .vdpCellContent {
+        background: LightBlue;
+    }
```
match blue/light blue theme for interactive components

cuahsimarko commented on hs_discover/public/index.html:
```diff
@@ -68,157 +85,7 @@ <h2 class="page-title">Discover
-                </div>
-            </div>
+            {% include 'includes/legend.html' %}
         </div>
```
modularize legend

cuahsimarko commented on hs_discover/src/components/Search.vue:
```diff
@@ -1,28 +1,24 @@
 <template>
-  <div id="discover-search">
+  <div id="discover-search" class="container">
     <input id="map-mode-button" type="button" class="btn btn-default mapdisp" value="Show Map" :disabled="!geoloaded"
```
bootstrap grid https://getbootstrap.com/docs/4.0/layout/grid/

cuahsimarko commented on hs_discover/src/components/Search.vue:
```diff
@@ -32,17 +28,20 @@
+                          <div class="date-wrapper">
+                            <span class="fa fa-calendar calendar-icon"></span>
+                              <date-pick style="width:167px"
                                  v-model="startdate"
```
keeping style outside of class is my informal convention to know to revisit when consolidating CSS files, scoped and other inline <style> that is sprawled all over the place, often with unused/stale code; temporary convention during ongoing cleanup

cuahsimarko commented on hs_discover/src/components/Search.vue:
```diff
@@ -184,24 +183,41 @@
-              ownerFilter.length || subjectFilter.length || contributorFilter.length || typeFilter.length ||
-              availabilityFilter.length || searchtext !== '' || startdate !== '' || enddate !== '')"><i>No resource matches</i></p>
+          <div class="row" style="padding-right:14px">
+
```
place paging controls above and below the resource table; controls will float below the filters in mobile/reduced window size use case

cuahsimarko commented on hs_discover/src/components/Search.vue:
```diff
@@ -184,24 +183,41 @@
-                            <th v-for="key in labels" v-bind:key="key" style="cursor:pointer"
-                                @click="sortBy(key)">
-                                <i :class="sortStyling(key)"></i>{{key}}
+                        <tr><th class="tbl-col-icons"><!-- placeholder --></th>
```
Instead of dynamic Vue table headers, use explicit headers, allowing assignment of classes with set widths. This is part of ongoing responsive design

cuahsimarko commented on hs_discover/src/components/Search.vue:
```diff
@@ -531,9 +558,48 @@ export default {
 
 <style scoped>
     @import url("https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css");
```
ids at top classes towards bottom

cuahsimarko commented on hs_discover/src/components/Search.vue:
```diff
@@ -567,53 +623,55 @@ export default {
+        margin-top: 0;
+    }
     .table-message {
-        position: absolute;
```
warnings of no resource matches moved above filters to serve mobile use case

cuahsimarko commented on theme/templates/includes/notifications-app.html:
```diff
@@ -1,3 +1,4 @@
+<script src="https://cdn.jsdelivr.net/npm/vue@2.6.12/dist/vue.js"></script>
 <li id="notifications-dropdown">
```
Provide a vue context to allow this to be independent for now, until future consolidation. Django includes are rendered independently then combined in the parent template.

hydrocheck commented:
Even with a switch to private browsing and clearing cache, the issue persists.  Here's a video from a VM:


https://user-images.githubusercontent.com/43424907/108139074-4aa59f80-708d-11eb-9559-28414d918ccb.mov

Castronova commented:
@cuahsimarko I just looked at https://dev-hs-6.cuahsi.org/ and overall it looks much better, thanks for taking this work on. Obviously it's not ideal to have so many facets filling up the screen but this is something we can deal with in a next step. How hard would it be to collapse all facets when the screen enters mobile view? This would improve how it looks quite a bit.

Castronova commented:
> Even with a switch to private browsing and clearing cache, the issue persists. Here's a video from a VM:
> 
>  Screen.Recording.2021-02-16.at.7.25.38.PM.mov

@hydrocheck I'm not experiencing this on Chrome Version 88.0.4324.150 (Official Build) (x86_64).

cuahsimarko commented:
@Castronova for mobile, I think facets/checkboxes should be re-thought. We can discuss another time.

cuahsimarko commented:
@hydrocheck can you send me more information about the macOS running inside of a virtual machine? Also, do you have access to native macOS / iPad /iPhone on Apple hardware?

dan-palmer1 commented:
I am running MacOS Catalina 10.15.6 on an MacBook.
Looking at the page in Firefox and Chrome the table shows up correctly. 
However in Safari 13.1.2 the table is found below the filters.  Resizing the window has no effect on the table location.

cuahsimarko commented:
The table issue seems to be only Safari and only certain versions of macOS. I'm on Big Sur with Safari 14. I thought I tested on High Sierra and I definitely tested on the iPad Safari a few times throughout the testing across Chrome, Firefox and Safari and mobile devices.

![image](https://user-images.githubusercontent.com/45397838/108259276-e509ef80-712e-11eb-990c-e8f3df7617a7.png)

I'll try a few things, but since I don't have access to other ways of testing, I'll have to iterate with the CUAHSI team. Martin and I will revisit this, as I also am unable to test on Microsoft. We will need a more comprehensive solution for testing and developer process. Right now, this all falls on me as I do all of this extensive cleanup, but at some point as others contribute more to frontend, expectations will be that they find ways of testing cross browser. For now, I'm using personal devices and those of friends.

cuahsimarko commented:
Any time I've tested this on Safari, it's fine. Actually any browser and OS and device all displayed fine, as mentioned.

cuahsimarko commented:
This is Safari 13 and 14 across several devices in emulator mode, none of which exhibit the problem. I'll find other software or borrow an old mac if I can find one still on Catalina. We should also lock down supported platforms....
![image](https://user-images.githubusercontent.com/45397838/108266033-94e35b00-7137-11eb-8ad8-8c655f8a7503.png)

cuahsimarko commented:
I am able to reproduce on this trial testing software. I can just keep opening trials indefinitely until we have a more solid policy and funding/priority for this direction. 
![image](https://user-images.githubusercontent.com/45397838/108267764-9f065900-7139-11eb-900e-dd20143e4582.png)

cuahsimarko commented:
Add to this hiding no resource matches during load

cuahsimarko commented:
I just tested on catalina, safari 14 and it's fine...

cuahsimarko commented:
Will redo work once access to Catalina with older Safari versions is available. Also, will need access to Microsoft/Windows latest and older versions.
